server/public/plugin/client.go

```diff
@@ -12,10 +14,7 @@ const (
 	BotUserKey        = InternalKeyPrefix + "botid"
 )
 
-// Starts the serving of a Mattermost plugin over net/rpc. gRPC is not yet supported.
-//
-// Call this when your plugin is ready to start.
-func ClientMain(pluginImplementation any) {
+func initializePluginImplementation(pluginImplementation any) map[string]plugin.Plugin {
```

Comment:
Oops, I pulled this out when I originally had a `ClientMain` and a `ClientMainTesting`, but moved to the options approach instead. Happy to inline or keep as feedback received.

Reply:
1/5 that there is no need to have a separate method and `initializePluginImplementation` should be inlined into `ClientMain`.